File: Data Structure/14425번 - 문자열 집합/14425.java
import java.io.*;
import java.util.*;

public class Main {
    static int n, m;
    static Set<String> setStr = new HashSet<>();
    static List<String> insStr = new ArrayList<>();

    public static void main(String[] args) throws IOException {
        input();
        solve();
    }

    static void input() throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;

        st = new StringTokenizer(br.readLine());
        n = Integer.parseInt(st.nextToken());
        m = Integer.parseInt(st.nextToken());

        for (int i = 0; i < n; i++) {
            setStr.add(br.readLine());
        }
        for (int i = 0; i < m; i++) {
            insStr.add(br.readLine());
        }
    }

    static void solve() {
        int ans = 0;

        for (String s : insStr) {
            if (setStr.contains(s)) ans++;
        }

        System.out.println(ans);
    }
}

File: Data Structure/14425번 - 문자열 집합/14425.py
import sys

input = sys.stdin.readline

def solve(set_str, ins_str):
    ans = 0
    for ins in ins_str:
        if ins in set_str:
            ans += 1
    
    print(ans)

if __name__ == '__main__':
    n, m = map(int, input().split())
    set_str = set([input().rstrip() for _ in range(n)])
    ins_str = [input().rstrip() for _ in range(m)]
    solve(set_str, ins_str)